flycheck.el

```diff
@@ -9890,9 +9895,21 @@ See URL `http://call-cc.org/'."
                                  (zero-or-more not-newline))
                    (one-or-more space) "<--")
           line-end)
+   ;; A of, version 4.12.0, the chicken compiler doesn't currently provide a
+   ;; line number for this error.
+   (error line-start "Syntax error: illegal atomic form\n\n"
+          (message (one-or-more not-newline)
+                   (zero-or-more "\n"
+                                 (zero-or-more space)
+                                 (zero-or-more not-newline))
+                   (one-or-more space) "<--")
+          line-end)
```

Comment:
The chicken compiler immediately crashes if a single empty `()` form is present in the buffer, thereby disabling flycheck entirely for the buffer. I added this form to at least ignore the noisy errors (this pops up every few seconds when coding with paredit), but I don't think this is the best way to proceed. Is there a way of preventing flycheck from updating if a `()` form is present in the buffer?

Reply:
Example compiler output:

```
Syntax error: illegal atomic form

	()

	Expansion history:

	library.scm:3158: ##sys#print	  
	library.scm:3188: case-sensitive	  
	library.scm:3189: keyword-style	  
	library.scm:3190: ##sys#print-length-limit	  
	library.scm:3281: outstr	  
	library.scm:3188: g5138	  
	extras.scm:630: loop	  
	extras.scm:621: newline	  
	library.scm:3154: ##sys#write-char/port	  
	library.scm:3146: ##sys#write-char-0	  
	library.scm:3139: g4941	  
	extras.scm:630: loop	  
	support.scm:122: print-call-chain	  
	library.scm:3882: ##sys#get-call-chain	  
	library.scm:3834: ##sys#make-vector	  
	library.scm:1371: ##sys#allocate-vector	  	<--

Error: shell command terminated with non-zero exit status 17920: '/usr/bin/chicken' 'test/resources/language/chicken/io-type-warning.scm' -output-file 'test/resources/language/chicken/io-type-warning.c' -analyze-only -local
```

Reply:
There seems to be a class of compiler errors that just exit immediately without producing usable output.

Here is another one:

```
Syntax error: malformed expression

	("foo" . "bar")

	Expansion history:

	library.scm:3348: outchr	  
	library.scm:3188: g5148	  
	library.scm:3327: outchr	  
	library.scm:3188: g5148	  
	library.scm:3359: outchr	  
	library.scm:3188: g5148	  
	extras.scm:630: loop	  
	extras.scm:621: newline	  
	library.scm:3154: ##sys#write-char/port	  
	library.scm:3146: ##sys#write-char-0	  
	library.scm:3139: g4941	  
	extras.scm:630: loop	  
	support.scm:122: print-call-chain	  
	library.scm:3882: ##sys#get-call-chain	  
	library.scm:3834: ##sys#make-vector	  
	library.scm:1371: ##sys#allocate-vector	  	<--

Error: shell command terminated with non-zero exit status 17920: '/usr/bin/chicken' 'test/resources/language/chicken/io-type-warning.scm' -output-file 'test/resources/language/chicken/io-type-warning.c' -analyze-only -local
```

Reply:
I think the good approach here is to match on all `Syntax error: ...`, not just the `illegal atomic form`.

Too bad `chicken` doesn't give a line/column information for these, as that would be more useful.